File: takeItEasy-domain/src/main/java/com/sudosoo/takeiteasy/service/CommentService.java
package com.sudosoo.takeiteasy.service;

import com.sudosoo.takeiteasy.dto.comment.CreateCommentRequestDto;
import com.sudosoo.takeiteasy.entity.Comment;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CommentService {
    Comment create(CreateCommentRequestDto createCommentRequestDto);
    Comment getByCommentId(Long commentId);
    Page<Comment> getCommentPaginationByPostId(Long postId, Pageable pageRequest);
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/service/CommentService.kt
package com.sudosoo.takeItEasy.service

import com.sudosoo.takeItEasy.dto.comment.CreateCommentRequestDto
import com.sudosoo.takeiteasy.entity.Comment
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface CommentService {
    fun create(createCommentRequestDto: CreateCommentRequestDto): Comment
    fun getByCommentId(commentId: Long): Comment
    fun getCommentPaginationByPostId(postId: Long, pageRequest: Pageable): Page<Comment>
}
